src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -53,467 +42,337 @@ export interface ComboBoxProps {
-            fullWidth
-            preventAutofocus
-          >
-            <Popover.Pane onScrolledToBottom={onEndReached}>
-              <div
-                id={this.state.comboBoxId}
-                role="listbox"
-                aria-multiselectable={allowMultiple}
-              >
-                {contentBefore}
-                {actionsBeforeMarkup}
-                {optionsMarkup}
-                {actionsAfterMarkup}
-                {contentAfter}
-                {emptyStateMarkup}
-              </div>
-            </Popover.Pane>
-          </Popover>
-        </div>
-      </ComboBoxContext.Provider>
-    );
-  }
+    let newIndex = selectedIndex;
 
-  private handleDownArrow = () => {
-    this.selectNextOption();
-    this.handlePopoverOpen;
-  };
+    if (selectedIndex <= 0) {
+      newIndex = navigableOptions.length - 1;
+    } else {
+      newIndex--;
+    }
 
-  private handleUpArrow = () => {
-    this.selectPreviousOption();
-    this.handlePopoverOpen;
-  };
+    selectOptionAtIndex(newIndex);
+  }, [navigableOptions, selectOptionAtIndex, selectedIndex]);
 
-  private handleEnter = (event: KeyboardEvent) => {
-    if (event.keyCode !== Key.Enter) {
-      return;
-    }
+  const selectOptions = useCallback(
+    (selected: string[]) => {
+      selected && onSelect(selected);
+      if (!allowMultiple) {
+        resetVisuallySelectedOptions();
+        forcePopoverActiveFalse();
+      }
+    },
+    [
+      allowMultiple,
+      forcePopoverActiveFalse,
+      onSelect,
+      resetVisuallySelectedOptions,
+    ],
+  );
 
-    const {selectedOption} = this.state;
-    if (this.state.popoverActive && selectedOption) {
-      if (isOption(selectedOption)) {
-        event.preventDefault();
-        this.handleSelection(selectedOption.value);
+  const handleSelection = useCallback(
+    (newSelected: string) => {
+      let newlySelectedOptions = selected;
+      if (selected.includes(newSelected)) {
+        newlySelectedOptions.splice(
+          newlySelectedOptions.indexOf(newSelected),
+          1,
+        );
+      } else if (allowMultiple) {
+        newlySelectedOptions.push(newSelected);
       } else {
-        selectedOption.onAction && selectedOption.onAction();
+        newlySelectedOptions = [newSelected];
       }
-    }
 
-    this.handlePopoverOpen;
-  };
-
-  private handleFocus = () => {
-    this.setState({popoverActive: true, popoverWasActive: true});
-  };
+      selectOptions(newlySelectedOptions);
+    },
+    [allowMultiple, selectOptions, selected],
+  );
 
-  private handleBlur = () => {
-    this.setState({popoverActive: false, popoverWasActive: false}, () => {
-      this.resetVisuallySelectedOptions();
-    });
-  };
+  const handleDownArrow = useCallback(() => {
+    selectNextOption();
+  }, [selectNextOption]);
 
-  private handleClick = () => {
-    !this.state.popoverActive && this.setState({popoverActive: true});
-  };
+  const handleUpArrow = useCallback(() => {
```

Comment:
`selectPreviousOption` is already a function wrapped in a useCallback so repeating that doesn't seem that useful

Down below, replace the following and delete this

```diff
- <KeypressListener keyCode={Key.UpArrow} handler={handleUpArrow} />
+ <KeypressListener keyCode={Key.UpArrow} handler={selectPreviousOption} />
```